my-app/src/Map.js

```diff
@@ -0,0 +1,129 @@
+import { useSearchParams } from 'react-router-dom';
+import React, { useState } from 'react';
+import { GoogleMap, useJsApiLoader , Marker} from "@react-google-maps/api";
+import './css/Map.css';
+
+
+
+
+function Map(){
+
+  const [latitude, setLatitude] = useState(0);
+  const [longitude, setLongitude] = useState(0);
+  const [searchParams] = useSearchParams();
+  
+  const nearbySearch = () => {
+    let currentPos = new window.google.maps.LatLng(latitude, longitude);
+    var service = new window.google.maps.places.PlacesService(document.getElementById('map2'));
```

Comment:
Nit: Apparently, var is scoped to the entire function body while let and const are scoped to the next enclosing bracket. So, let's try to reduce the scope of the variable as much as possible and use immutable variables if we can.